tests/UnitTests/ApiConnectors/BaseApiConnectorTest.php

```diff
@@ -286,4 +285,119 @@ public function testResendXmlDocumentStopsAfterMaxRetries(\Exception $exception)
         $numRetries = $this->getObjectAttribute($this->service, "numRetries");
         $this->assertEquals(0, $numRetries);
     }
+    /**
+     * @dataProvider soapFaultProvider
+     * @dataProvider errorExceptionProvider
+     */
+    public function testResendXmlDocumentStopsAfterCustomMaxRetries(\Exception $exception)
+    {
+        $this->service = $this->getMockBuilder(BaseApiConnector::class)
+            ->setConstructorArgs(
+                [
+                    $this->connection,
+                    [
+                        BaseApiConnector::CONFIG_MAX_RETRIES => 5
+                    ]
+                ]
+            )
+            ->getMockForAbstractClass();
+
+        $this->connection->expects($this->exactly($this->getMaxNumRetries() + 1))
+            ->method("getAuthenticatedClient")
+            ->with(Services::PROCESSXML())
+            ->willReturn($this->client);
+
+        $this->connection->expects($this->exactly($this->getMaxNumRetries() + 1))
+            ->method("resetClient")
+            ->with(Services::PROCESSXML());
+
+        $this->client->expects($this->exactly($this->getMaxNumRetries() + 1))
+            ->method("sendDocument")
+            ->willThrowException($exception);
+
+        $this->expectException(Exception::class);
+        $this->service->sendXmlDocument(new \DOMDocument());
+
+        $numRetries = $this->getObjectAttribute($this->service, "numRetries");
```

Comment:
This is never run and should be removed

Reply:
I checked the code and after there are no more retries left. It will set the numRetries to zero. For me, it does not make any sense. But this test is passing.